G/gmsh/build_tarballs.jl

```diff
@@ -30,22 +34,56 @@ sed ${prefix}/lib/gmsh.jl.bak \
   -e 's/^\(const lib.*\)/#\1/g' \
   -e 's/^\(module gmsh\)$/\1\nusing gmsh_jll: libgmsh\nconst lib = libgmsh/g' \
   > ${prefix}/lib/gmsh.jl
+install_license ../LICENSE.txt
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = expand_cxxstring_abis(supported_platforms())
+filter!(p -> !(Sys.iswindows(p) && cxxstring_abi(p) == "cxx03"), platforms) # Restriction from OCCT builds
```

Comment:
I had missed this:
https://github.com/JuliaPackaging/Yggdrasil/blob/892c5abc72b0c74f864732ea437e477419f37774/O/OCCT/build_tarballs.jl#L47
and I don't understand it.  Please see https://docs.binarybuilder.org/stable/building/#Expanding-C-string-ABIs-or-libgfortran-versions

> This ABI does _not_ have to do with the C++ standard used by the source code, in fact you can build a C++03 library with the C++11 `std::string` ABI and a C++11 library with the C++03 `std::string` ABI. This is achieved by appropriately setting the `_GLIBCXX_USE_CXX11_ABI` macro.

Reply:
This was something I noticed when building OCCT in #4888. For windows builds, I got errors about from the code using std::swap except when disabling the C++03 ABI. I can look back into it a little more.

Reply:
It'd be good to see _what_ is exactly the error.  And again, the C++ string ABI is completely different from the standard used in the source code.  Since the result here would be the narrowing of the supported platforms for an existing platform, this isn't a desirable outcome.

Reply:
Actually, I think I was testing incorrectly and this actually is not required you are right, I'm sorry about that. Fixed in #4911 and here in the most recent update.

Reply:
Thanks for your help. I think we need to somehow rerun the Windows C++03 pipelines to use the new OpenCASCADE dependency build, but otherwise things look good. Thank you again!